cloudbuild/cloudbuild.yaml

```diff
@@ -22,13 +22,44 @@ steps:
     id: 'unit-tests-hadoop3'
     waitFor: ['docker-build']
     entrypoint: 'bash'
-    args: ['/bigdata-interop/cloudbuild/presubmit.sh', 'hadoop3']
+    args: ['/bigdata-interop/cloudbuild/presubmit.sh', 'hadoop3', 'unittest']
     env:
       - 'CODECOV_TOKEN=$_CODECOV_TOKEN'
       - 'VCS_BRANCH_NAME=$BRANCH_NAME'
       - 'VCS_COMMIT_ID=$COMMIT_SHA'
       - 'VCS_TAG=$TAG_NAME'
       - 'CI_BUILD_ID=$BUILD_ID'
-    
+
+  # 4. Run Hadoop 2 integration tests concurrently
+  - name: 'gcr.io/$PROJECT_ID/dataproc-bigdata-interop-presubmit'
+    id: 'integration-tests-hadoop2'
+    waitFor: ['docker-build']
+    entrypoint: 'bash'
+    args: ['/bigdata-interop/cloudbuild/presubmit.sh', 'hadoop2', 'integrationtest']
+    env:
+      - 'GCS_TEST_PROJECT_ID=$PROJECT_ID'
+      - 'CODECOV_TOKEN=$_CODECOV_TOKEN'
+      - 'VCS_BRANCH_NAME=$BRANCH_NAME'
+      - 'VCS_COMMIT_ID=$COMMIT_SHA'
+      - 'VCS_TAG=$TAG_NAME'
+      - 'CI_BUILD_ID=$BUILD_ID'
+
+  # 5. Run Hadoop 3 integration tests sequentially
+  - name: 'gcr.io/$PROJECT_ID/dataproc-bigdata-interop-presubmit'
+    id: 'integration-tests-hadoop3'
+    waitFor: ['integration-tests-hadoop2']
+    entrypoint: 'bash'
+    args: ['/bigdata-interop/cloudbuild/presubmit.sh', 'hadoop3', 'integrationtest']
+    env:
+      - 'GCS_TEST_PROJECT_ID=$PROJECT_ID'
+      - 'CODECOV_TOKEN=$_CODECOV_TOKEN'
+      - 'VCS_BRANCH_NAME=$BRANCH_NAME'
+      - 'VCS_COMMIT_ID=$COMMIT_SHA'
+      - 'VCS_TAG=$TAG_NAME'
+      - 'CI_BUILD_ID=$BUILD_ID'
+
+timeout: 1800s
```

Comment:
This is too large, can you add a comment on how much time it typically takes?

Reply:
Done.